### PR TITLE
Fix: mongoDB datastore can't list special email user(#4104)

### LINKED ISSUE #4104: use mongoDB datastorage to integration sso fail
**Describe the bug**

我希望使用dex插件集成ldap来实现单点登录的功能，按照官方的文档进行配置并已经将admin账号的邮箱设置为和ldap账号邮箱相同
当使用ldap中和admin配置的邮箱相同的账号登录时，并没有继承原本的admin的,并且在登出再次登录同样的账号时页面显示“data record is exist”，但是我将存储方式改为kubeAPI后则可以正常的使用并可以继承原本本地账号的权限

I hope use dex addon integration ldap to implementa sso function,i had config admin account email same as ldap account.When i login ,this account not inheritance the local admin account permissions,and when i login by same account again ,the page display "data record is exist",but when i change the datastorage to kubeapi,all thing become right .

**To Reproduce**

**Expected behavior**

使用ldap账号可以登录kubevela并且相同邮箱的账号可以继承本地账号的权限
use ldap account login kubevela and the same email account can inheritance the permissions
**Screenshots**

企业微信截图_16540766817109

企业微信截图_16540817044590

**KubeVela Version**
kubevela core 1.4.0
kubevela cli 1.4.0
velaux 1.4.0

**Cluster information**

docker info
```
Client: Docker Engine - Community
 Version:           20.10.16

Server: Docker Engine - Community
 Engine:
  Version:          20.10.0
```
kubernetes info
```
Kubernetes v1.21.0
```
mongoDB
```
mongo:4.2.9
```

**Additional context**

### PR DESCRIPTION
**Pull request by fengkang01**
Signed-off-by: fengkang <fengkangb@digitalchina.com>

### Description of your changes
I change the user model index function ,and replace symbol timing.
In the kubeapi datastore , the list function can get include Special characters("@"," ")filed "name" and "email",i use verifyUserValue function to replace the Special characters to make sure that can be used by k8s label
In the mongoDB datastore , this list funciton keep intact,so that the program can special a email to list the model user

Fixes #4104 

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [x] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### Special notes for your reviewer
@FogDong

### CODE REVIEW

File: pkg/apiserver/infrastructure/datastore/kubeapi/kubeapi.go [hunk 1]
```diff
@@ -473,3 +479,9 @@ func (m *kubeapi) Count(ctx context.Context, entity datastore.Entity, filterOpti
 	}
 	return int64(len(items)), nil
 }
+
+func verifyUserValue(v string) string {
```
Comment (FogDong) [inline pkg/apiserver/infrastructure/datastore/kubeapi/kubeapi.go:483]:
```suggestion
func verifyValue(v string) string {
```

File: pkg/apiserver/infrastructure/datastore/kubeapi/kubeapi.go [hunk 2]
```diff
@@ -84,6 +84,9 @@ func (m *kubeapi) generateConfigMap(entity datastore.Entity) *corev1.ConfigMap {
 	if labels == nil {
 		labels = make(map[string]string)
 	}
+	for k, v := range labels {
```
Comment (barnettZQG) [inline pkg/apiserver/infrastructure/datastore/kubeapi/kubeapi.go:87]: It would be better to add some test cases.

Comment (fengkang01) [reply pkg/apiserver/infrastructure/datastore/kubeapi/kubeapi.go:87]: i will add some test cases to test the verifyValue function's effectiveness

**Review by barnettZQG — APPROVED** [review]
LGTM

**Review by FogDong — APPROVED** [review]

**Review by yangsoon — APPROVED** [review]

**Review by zzxwill — APPROVED** [review]

**Pull request merged**